app/experimenter/legacy-ui/rapid/components/experiments/ResultsTable.tsx

```diff
@@ -0,0 +1,270 @@
+  const analysisData = data.analysis;
+  const fullResultsData = analysisData["overall"];
+
+  const results = {};
+  let totalPopulation = 0;
+  Object.values(fullResultsData).forEach((row: AnalysisPoint) => {
+    const { metric, branch, statistic, point, lower, upper, comparison } = row;
+    results[branch] = results[branch] || {};
+
+    // `comparison` could be empty, `difference`, or `relative_uplift`.
+    // Here we want to display the empty comparison which means a given
+    // branch's values is not relative to another branch. We also want
+    // to use the `difference` comparison to decide whether there was
+    // any positive or negative significance.
+    if (metric in resultMetrics && resultMetrics[metric] === statistic) {
+      results[branch][metric] = results[branch][metric] || {};
+
+      let newData = {};
+      if (!comparison) {
+        newData = { lower, upper, point };
+      }
+
+      if (comparison === "difference") {
+        newData = { significance: computeSignificance(lower, upper) };
+      }
+
+      Object.assign(results[branch][metric], newData);
+    }
+
+    // Compute the population sum to be used for % per variant.
+    if (metric === "identity") {
+      totalPopulation += point;
+    }
+  });
+  setPopulationPercentages(results, totalPopulation);
+
+  return results;
+};
+
+const populationField = (point, percent) => {
+  return (
+    <div>
+      <div className="font-weight-bold">{point}</div>
+      <div className="subscript">{percent}%</div>
+    </div>
+  );
+};
+
+const showSignificanceField = (significance, interval) => {
+  {
+    let significanceIcon, className;
+    switch (significance) {
+      case Significance.POSITIVE:
+        significanceIcon = (
+          <i
+            className="fas fa-arrow-up"
+            data-tip={SIGNIFICANCE_TIPS.POSITIVE}
+          />
+        );
+        className = "positive-significance";
+        break;
+      case Significance.NEGATIVE:
+        significanceIcon = (
+          <i
+            className="fas fa-arrow-down"
+            data-tip={SIGNIFICANCE_TIPS.NEGATIVE}
+          />
+        );
+        className = "negative-significance";
+        break;
+      case Significance.NEUTRAL:
+        significanceIcon = (
+          <i className="fas fa-minus" data-tip={SIGNIFICANCE_TIPS.NEUTRAL} />
+        );
+        className = "neutral-significance";
+        break;
+    }
+
+    return (
+      <div className={className}>
+        <ReactTooltip />
+        {significanceIcon}
+        &nbsp;
+        {interval}
+      </div>
+    );
+  }
+};
+
+const countField = (lower, upper, significance) => {
+  const interval = (
+    <>
+      {Math.round(lower)} to {Math.round(upper)}
+    </>
+  );
+  return showSignificanceField(significance, interval);
+};
+
+const percentField = (lower, upper, significance) => {
+  const interval = (
+    <>
+      {Math.round(lower * 1000) / 10}% to {Math.round(upper * 1000) / 10}%
+    </>
```

Comment:
Is there a specific reason this and the above need to be JSX, or could they just be template/interpolated strings?

Reply:
nope. thanks